src/RefractAPI.cc

```diff
@@ -399,23 +461,26 @@ namespace drafter {
         return element;
     }
 
-    refract::IElement* ResourceToRefract(const snowcrash::Resource& resource)
+    refract::IElement* ResourceToRefract(const SectionInfo<snowcrash::Resource>& resource)
     {
         refract::ArrayElement* element = new refract::ArrayElement;
         RefractElements content;
 
         element->element(SerializeKey::Resource);
-        element->meta[SerializeKey::Title] = refract::IElement::Create(resource.name);
-        element->attributes[SerializeKey::Href] = refract::IElement::Create(resource.uriTemplate);
 
-        if (!resource.parameters.empty()) {
-            element->attributes[SerializeKey::HrefVariables] = ParametersToRefract(resource.parameters);
+        element->meta[SerializeKey::Title] = PrimitiveToRefract(MAKE_SECTION_INFO(resource, name));
+
+        element->attributes[SerializeKey::Href] = PrimitiveToRefract(MAKE_SECTION_INFO(resource, uriTemplate));
+
+        if (!resource.section.parameters.empty()) {
+            element->attributes[SerializeKey::HrefVariables] = ParametersToRefract(MAKE_SECTION_INFO(resource, parameters));
         }
 
-        content.push_back(CopyToRefract(resource.description));
-        content.push_back(DataStructureToRefract(resource.attributes));
+        content.push_back(CopyToRefract(MAKE_SECTION_INFO(resource, description)));
+        content.push_back(DataStructureToRefract(MAKE_SECTION_INFO(resource, attributes)));
 
-        std::transform(resource.actions.begin(), resource.actions.end(), std::back_inserter(content), ActionToRefract);
+        SectionInfoCollection<snowcrash::Actions> actions(resource.section.actions, resource.sourceMap.actions);
+        std::transform(actions.begin(), actions.end(), std::back_inserter(content), ActionToRefract);
```

Comment:
How about something like this?

```cpp
#define NODE_INFO(from, member) from.section.member, from.section.member

template<typename T>
void SomeFnName(const T& node, const SourceMap<T>& sourceMap, RefractElements content, Functor &wrapper) {
    NodeInfoCollection<T> collection(node, sourceMap);
    std::transform(collection.begin(), collection.end(), std::back_inserter(content), wrapper);
}
```

Use it as

```cpp
SomeFnName(NODE_INFO(resource, actions), content, ActionToRefract);
```

You can also modify the `MAKE_NODE_INFO` macro to use `NODE_INFO`